src/snyk/advisor/editor/EditorDecorator.ts

```diff
@@ -0,0 +1,75 @@
+import * as vscode from 'vscode';
```

Comment:
Please eleborate.

Reply:
When you write unit tests, "vscode" module cannot be resolved since tests run not within VS Code. To overcome this, we invert our dependency on vscode module through dependency inversion mechanisms. You can see inversions in `src/snyk/common/vscode` folder.